Resources/public/css/theme/modules/form.css

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+
+.pure-g [class *= "pure-u"],
+.yui3-g [class *= "yui3-u"],
+button,
+input,
+textarea {
+    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
+}
+
+a.ez-button {
```

Comment:
same here, not really related to forms. If you really want to move it, you can put in theme/modules/button.css (in a separate commit)